packages/shared/sierra-client/src/index.ts

```diff
@@ -77,57 +105,84 @@ export default class SierraClient {
       },
       validateStatus: status => status === 200
     }).then(response => {
-      return axios.create({
+      const instance = axios.create({
         baseURL: this.apiRoot,
         headers: {
           Authorization: 'Bearer ' + response.data.access_token
-        },
+        }
       });
+      instance.interceptors.response.use(response => response, error => this.unhandledError(error));
```

Comment:
Errors are turned into whatever `this.unhandledError(...)` is returning here, but the error is never caught by checking the promise for errors.

Reply:
That's not quite right. This construct here is defining the default handler for anything which Axios construes as an erroneous response, i.e. request errors such as timeouts, or any response with an HTTP status code that doesn't match the `validateStatus` function. When such errors occur, the caller will receive our custom `ErrorResponse` type, with a `status` of `SierraStatus.UnknownError` and a descriptive error message in the `message` field.

There's no error / exceptions being thrown around here, so `try / catch` or `Promise`'s `.catch(...)` doesn't apply.

Reply:
With that being said, there is a problem here - `this.unhandledError(error)` will receive an `AxiosError`, not an `AxiosResponse`. Will fix that.

Reply:
I've updated the status code / error handling approach to resolve this.